### PR TITLE
Add [extras] + [targets]

### PR DESCRIPTION
**Pull request by paraynaud**

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -19,3 +19,15 @@ Knet = "1.4.9"
...
 NLPModels = "0.18, 0.19"
+
+[extras]
+CUDA = "052768ef-5323-5732-b1bb-66c8b64840ba"
+IterTools = "c8e1da08-722c-5040-9ed9-7db0dc04731e"
+Knet = "1902f260-5fb4-5aff-8c31-6271790ab950"
+MLDatasets = "eb30cadb-4394-5ae3-aed4-317e484a6458"
+NLPModels = "a4795742-8479-5a88-8948-cc11e1c8c1a6"
+Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
+Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
```
Comment (dpo) [inline Project.toml:30]: Don't repeat dependencies. But I don't thin CUDA, etc, are dependencies, so they should be removed from above.

**Review by dpo — CHANGES_REQUESTED** [review]

**Comment by paraynaud** [comment]
I removed the packages not needed.
They are still needed in the tests which basically:
- create a Knet neural network;
- form a KnetNLPModel with it and a dataset;
- evaluate `obj` and `grad` from a KnetNLPModel.

**Comment by dpo** [comment]
It's much lighter now!

**Review by dpo — APPROVED** [review]

**Comment by paraynaud** [comment]
The tests run on GPU, I merge-rebase.

**Pull request merged**